storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorageTest.java

```diff
@@ -399,20 +403,21 @@ public Verifier(final LocalTieredStorage remoteStorage, final TopicIdPartition t
             this.topicIdPartition = requireNonNull(topicIdPartition);
         }
 
-        private List<Path> expectedPaths(final RemoteLogSegmentId id) {
+        private List<Path> expectedPaths(final RemoteLogSegmentMetadata metadata) {
             final String rootPath = getStorageRootDirectory();
             TopicPartition tp = topicIdPartition.topicPartition();
             final String topicPartitionSubpath = format("%s-%d-%s", tp.topic(), tp.partition(),
                     topicIdPartition.topicId());
-            final String uuid = id.id().toString();
+            final String uuid = metadata.remoteLogSegmentId().id().toString();
+            final long startOffset = metadata.startOffset();
 
             return Arrays.asList(
-                    Paths.get(rootPath, topicPartitionSubpath, uuid + LogFileUtils.LOG_FILE_SUFFIX),
-                    Paths.get(rootPath, topicPartitionSubpath, uuid + LogFileUtils.INDEX_FILE_SUFFIX),
-                    Paths.get(rootPath, topicPartitionSubpath, uuid + LogFileUtils.TIME_INDEX_FILE_SUFFIX),
-                    Paths.get(rootPath, topicPartitionSubpath, uuid + LogFileUtils.TXN_INDEX_FILE_SUFFIX),
-                    Paths.get(rootPath, topicPartitionSubpath, uuid + LEADER_EPOCH_CHECKPOINT.getSuffix()),
-                    Paths.get(rootPath, topicPartitionSubpath, uuid + LogFileUtils.PRODUCER_SNAPSHOT_FILE_SUFFIX)
+                    Paths.get(rootPath, topicPartitionSubpath, startOffset + "-" + uuid + LogFileUtils.LOG_FILE_SUFFIX),
```

Comment:
Ideally, we want the test implementation to be as close to the actual log file implementation as possible. Considering that, could we use `LogFileUtils#logFile(File dir, long offset)` here? Same for index file names.

Reply:
@divijvaidya Thanks for your feedback. I think the actual log file was actually named as [offset].log. Looking at the implementation of `LogFileUtils#logFile(File dir, long offset)`, I don't think it will allow us to insert a uuid in the middle as part of the filename. 

If we are to keep the `[offset-uuid.filetype]` pattern, instead of using `LogFileUtils#logFile(File dir, long offset)`, maybe we should make `LogFileUtils#filenamePrefixFromOffset(long offset)` as a public method so that we can construct a real offset using this method. What do you think ? 

FYI, the method to create these offloaded files is `RemoteLogSegmentFileset#openFileset(final File storageDir, final RemoteLogSegmentId id)` . Currently my PR has changed this method to accept `RemoteLogSegmentMetadata` instead of `RemoteLogSegmentId` , get offset from metadata, and prepend it to the filename. (So yes, it's not close to the actual log file implementation, as the offset was just "0" without formatting, instead of "0000000")

Reply:
> I don't think it will allow us to insert a uuid in the middle as part of the filename.

Ack. I missed that.

> maybe we should make LogFileUtils#filenamePrefixFromOffset(long offset) as a public method so that we can construct a real offset using this method. What do you think ?

Yes please. Let's use that.

Reply:
@divijvaidya I've changed the code to use `LogFileUtils#filenamePrefixFromOffset(long offset)`. The filename now should look like a real log file implementation like `00000000000000000011-oAtiIQ95REujbuzNd_lkLQ.log`